metaflow/datastore/spin_datastore/spin_inputs_datastore.py

```diff
@@ -0,0 +1,53 @@
+from itertools import chain
+
+
+class SpinInputsDatastore(object):
+    def __init__(self, inp_datastores, artifacts={}):
+        # This is a linear step, so we only have one input datastore
+        self._inp_datastores = inp_datastores
+        self._artifacts = artifacts
+        self._step_name = self._inp_datastores.step_name
+
+        # Set them to empty dictionaries in order to persist artifacts
+        # See `persist` method in `TaskDatastore` for more details
+        self._data = {}
+        self._objects = {}
+        self._info = {}
+
+    def __contains__(self, name):
+        try:
+            _ = self.__getattr__(name)
+        except AttributeError:
+            return False
+        return True
+
+    def __getitem__(self, name):
+        return self.__getattr__(name)
+
+    def __setitem__(self, name, value):
+        self._data[name] = value
+
+    def __getattr__(self, name):
```

Comment:
So we can do SID.bar for reading just like SID["bar"] but we can only set using SID["bar"]?